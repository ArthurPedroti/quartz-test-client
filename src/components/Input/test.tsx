import { render, screen } from '@testing-library/react'
import { Input } from '.'

describe('<Input />', () => {
  it('should render with passed values', () => {
    render(<Input name="teste" label="Deu certo" />)

    expect(screen.getByLabelText('Deu certo')).toBeInTheDocument()
  })

  it('should render witch passed value in "name"', () => {
    render(<Input name="teste" label="Deu certo" />)

    expect(screen.getByRole('textbox', { name: 'Deu certo' })).toHaveAttribute(
      'name',
      'teste'
    )
  })

  it('should render the chakra styles', () => {
    const { container } = render(<Input name="teste" label="Deu certo" />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
