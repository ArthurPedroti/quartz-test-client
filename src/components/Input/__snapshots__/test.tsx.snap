// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render the chakra styles 1`] = `
<div
  class="css-0"
>
  <div
    class="chakra-form-control css-0"
    role="group"
  >
    <label
      class="chakra-form__label css-uvho5g"
      for="teste"
      id="field-3-label"
    >
      Deu certo
    </label>
    <input
      class="chakra-input css-0"
      id="teste"
      name="teste"
    />
  </div>
</div>
`;
